api/v1/applicationlayer_types.go

```diff
@@ -83,6 +90,75 @@ type LogCollectionSpec struct {
+	// to run the WAF service on a different port, this field can be configured accordingly.
+	// It is important to ensure that no two services are running on the same port to avoid
+	// port conflicts.
+	// +optional
+	ListenPort int `json:"listenPort,omitempty"`
+	// CoreRuleSetsConfigMap specifies the name of the Kubernetes ConfigMap containing customized
+	// Web Application Firewall (WAF) core rule sets. If users want to customize or add WAF core rule sets,
+	// they need to create a Kubernetes ConfigMap and specify its name using this field. The ConfigMap should
+	// include the core ruleset files that will be used by Dikastes for executing the WAF functionality.
+	// This field is optional, and if not specified, no customized WAF core rule set will be enabled.
+	// +optional
+	CoreRuleSetsConfigMap string `json:"coreRuleSetsConfigMap,omitempty"`
+	// Workloads is a mandatory field that specifies the list of workloads for which the Web Application
+	// Firewall (WAF) needs to be enabled. Each workload is defined by the Workload struct, which includes
+	// details such as the name, namespace, context and specific labels for identifying the workload. The WAF
+	// functionality will be applied to these specified workloads to enhance their ingress traffic security.
+	Workloads []Workload `json:"workloads"`
+}
+
+type Workload struct {
+	// Name specifies the name of the EnvoyFilter which will direct the traffic to Dikastes for consulting the Web
+	// Application Firewall (WAF) functionality. The tigera-operator will automatically prefix this name with "tigera."
+	// and suffix it with ".waf-ext-authz" to signify that it is controlled by Tigera and used for external authorization
+	// for WAF. This name should be unique, and an ideal way to construct a name that ensures uniqueness is to follow the
+	// pattern "<namespace>.<context>.<application-name>". This naming convention helps in effectively identifying
+	// and managing the EnvoyFilters in the context of WAF.
+	Name string `json:"name"`
+	// Namespace specifies the Kubernetes namespace for which the EnvoyFilter will be applicable. This determines the
+	// scope of the EnvoyFilter's effect within the Kubernetes cluster. If the specified namespace is the Istio
+	// rootNamespace (i.e., the namespace in which Istio is installed), then the EnvoyFilter becomes a global filter,
+	// applying to workloads across all namespaces, provided their labels match the specified criteria.
+	Namespace string `json:"namespace"`
+	// Context specifies the context of the EnvoyFilter within the Istio service mesh. The value of this field will
+	// always be "gateway" for the Istio Gateway (either ingress or egress) and "sidecar" for Istio workloads. This
+	// field is crucial in determining the operational context of the EnvoyFilter, whether it is meant to be applied
+	// to traffic flowing through the Istio Gateway or to traffic within the individual Istio workloads (sidecars).
+	Context string `json:"context"`
```

Comment:
Need to add enum validation for context.
// +kubebuilder:validation:Enum=gateway;sidecar